File: share/pegasus/examples/simple-examples/merge.java
import edu.isi.pegasus.planner.dax.*;

public class merge {

    public static void main(String[] args) {
        ADAG dax = new ADAG("merge");

        String[] dirs = new String[] { "/bin","/usr/bin","/usr/local/bin" };
        Job[] jobs = new Job[3];
        File[] files = new File[3];

        for (int i=0; i<3; i++) {
            String dir = dirs[i];
            Job ls = new Job("ID"+i, "ls");
            jobs[i] = ls;
            ls.addArgument("-l").addArgument(dir);
            File f = new File("bin_"+i+".txt");
            files[i] = f;
            ls.uses(f, File.LINK.OUTPUT, File.TRANSFER.FALSE, false);
            ls.setStdout(f);
            dax.addJob(ls);
        }

        Job cat = new Job("ID3", "cat");
        for (File f : files) {
            cat.addArgument(f);
            cat.uses(f, File.LINK.INPUT);
        }
        File output = new File("binaries.txt");
        cat.setStdout(output);
        cat.uses(output, File.LINK.OUTPUT, File.TRANSFER.TRUE, false);
        dax.addJob(cat);

        for (Job j : jobs) {
            dax.addDependency(j, cat);
        }

        dax.writeToSTDOUT();
    }
}

File: share/pegasus/examples/simple-examples/merge.py
#!/usr/bin/env python3

import sys
from Pegasus.DAX3 import *

dax = ADAG("merge")

dirs = ["/bin","/usr/bin","/usr/local/bin"]
jobs = []
files = []

for i,d in enumerate(dirs):
    ls = Job("ls")
    jobs.append(ls)
    ls.addArguments("-l",d)
    f = File("bin_%d.txt" % i)
    files.append(f)
    ls.setStdout(f)
    ls.uses(f, link=Link.OUTPUT)
    dax.addJob(ls)

cat = Job("cat")
cat.addArguments(*files)
for f in files:
    cat.uses(f, link=Link.INPUT)
output = File("binaries.txt")
cat.setStdout(output)
cat.uses(output, link=Link.OUTPUT, transfer=True, register=False)
dax.addJob(cat)

for j in jobs:
    dax.depends(cat, j)

dax.writeXML(sys.stdout)

